reflex/app.py

```diff
@@ -161,11 +161,12 @@ def __init__(self, *args, **kwargs):
                     "rx.BaseState cannot be subclassed multiple times. use rx.State instead"
                 )
 
-            # verify that provided state is valid
-            if self.state and self.state is not State:
-                console.warn(
-                    f"Using substate ({self.state.__name__}) as root state in `rx.App` is currently not supported."
-                    f" Defaulting to root state: ({State.__name__})"
+            if "state" in kwargs:
+                console.deprecate(
+                    feature_name="`state` argument for App()",
+                    reason="due to multiple rx.State subclasses being allowed.",
```

Comment:
Should we more explicitly mention that since v0.2.x the substates are automatically inferred. So that people with older apps who weren't getting warnings before know why it's to required.